### PR TITLE
PF-635: relax creation parameters requirements for buckets

### PR DESCRIPTION
**Pull request by jaycarlton**
Previously we assumed the storage class and lifecycle rules were required, but they're actually not.

### CODE REVIEW

**Comment by jaycarlton** [comment]
Unit test failures should be fixed by https://github.com/DataBiosphere/terra-workspace-manager/pull/335

File: src/main/java/bio/terra/workspace/db/ResourceDao.java [hunk 1]
```diff
@@ -454,6 +454,7 @@ private void storeResource(WsmResource resource) {
             .addValue("attributes", resource.attributesToJson());
     if (resource.getStewardshipType().equals(CONTROLLED)) {
       ControlledResource controlledResource = resource.castToControlledResource();
+      //noinspection deprecation
```
Comment (ddietterich) [inline src/main/java/bio/terra/workspace/db/ResourceDao.java:457]: What does that mean?

**Review by ddietterich — APPROVED** [review]
lgtm

File: src/main/java/bio/terra/workspace/db/ResourceDao.java (hunk 1 above)
Comment (jaycarlton) [reply src/main/java/bio/terra/workspace/db/ResourceDao.java:457]: It shushes the deprecation warning. My thinking ATM is that I want to warn on new uses and grandfather in the two existing ones.

**Pull request merged**